modulemd/v2/modulemd-module-stream-v1.c

```diff
@@ -873,6 +873,120 @@ modulemd_module_stream_v1_get_xmd (ModulemdModuleStreamV1 *self)
+  v1_self_2 = MODULEMD_MODULE_STREAM_V1 (self_2);
+
+  if (!MODULEMD_MODULE_STREAM_CLASS (modulemd_module_stream_v1_parent_class)
+         ->equals (self_1, self_2))
+    {
+      return FALSE;
+    }
+
+  /*Check property equality*/
+  if (g_strcmp0 (v1_self_1->community, v1_self_2->community) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v1_self_1->description, v1_self_2->description) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v1_self_1->documentation, v1_self_2->documentation) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v1_self_1->summary, v1_self_2->summary) != 0)
+    return FALSE;
+
+  if (g_strcmp0 (v1_self_1->tracker, v1_self_2->tracker) != 0)
+    return FALSE;
+
+  if (!modulemd_buildopts_equals (v1_self_1->buildopts, v1_self_2->buildopts))
+    return FALSE;
+
+  if (!modulemd_hash_table_equals (v1_self_1->rpm_components,
+                                   v1_self_2->rpm_components,
+                                   modulemd_component_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v1_self_1->module_components,
+                                   v1_self_2->module_components,
+                                   modulemd_component_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v1_self_1->module_licenses,
+                                           v1_self_2->module_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v1_self_1->content_licenses,
+                                           v1_self_2->content_licenses))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v1_self_1->profiles,
+                                   v1_self_2->profiles,
+                                   modulemd_profile_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v1_self_1->rpm_api,
+                                           v1_self_2->rpm_api))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v1_self_1->rpm_artifacts,
+                                           v1_self_2->rpm_artifacts))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_sets_are_equal (v1_self_1->rpm_filters,
+                                           v1_self_2->rpm_filters))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (v1_self_1->servicelevels,
+                                   v1_self_2->servicelevels,
+                                   modulemd_service_level_equals_wrapper))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (
+        v1_self_1->buildtime_deps, v1_self_2->buildtime_deps, g_str_equal))
+    {
+      return FALSE;
+    }
+
+  if (!modulemd_hash_table_equals (
+        v1_self_1->runtime_deps, v1_self_2->runtime_deps, g_str_equal))
+    {
+      return FALSE;
+    }
+
+  if (g_variant_compare (v1_self_1->xmd, v1_self_2->xmd) != 0)
```

Comment:
`g_variant_compare()` won't work here. These aren't guaranteed to be basic types (in fact, they are pretty much guaranteed *not* to be).

From the [docs](https://developer.gnome.org/glib/stable/glib-GVariant.html#g-variant-compare): "It is a programmer error to attempt to compare container values or two values that have types that are not exactly equal. For example, you cannot compare a 32-bit signed integer with a 32-bit unsigned integer. Also note that this function is not particularly well-behaved when it comes to comparison of doubles; in particular, the handling of incomparable values (ie: NaN) is undefined.

If you only require an equality comparison, [`g_variant_equal()`](https://developer.gnome.org/glib/stable/glib-GVariant.html#g-variant-equal) is more general."